### DISCUSSION TITLE
Предложения

### DISCUSSION THREAD

**Discussion by bigmack2304** [OGSR/OGSR-Engine#311] [Ideas]
Уже давненько строю мод на базе вашего движка и за это время накопилось несколько предложений.

1.  Возможно-ли добавить параметр интенсивности солнечных лучей (по аналогии с командой r_sunshafts_intensity) в секцию с погодой ? Как это было в огсе. Ну а если нужного параметра в секции не будет то расчет интенсивности делался-бы по старому алгоритму (как сейчас).

2. Сейчас есть такая команда "r2_dof_far" которая устанавливает дистанцию, дальше которой все заблюривается dof'ом. Я предлагаю добавить в external.ltx параметр, включив который эта команда настраивалась-бы автоматически в зависимости от параметра far_plane из секции с погодой.  К примеру если far_plane = 800 то r2_dof_far = 65% от текущего far_plane,  r2_dof_far = 520 (на мой взгляд 65% это наиболее сбалансированно)

3. Еще предлагаю добавить возможность включать смерть от первого лица (в том-же external.ltx к примру), еще давно я хотел адаптировать эту правку 
```
--[[ ---------------------------------------------
 File       : sk_actor_dead.script
 Description: Смерть ГГ от первого лица
 Copyright  : 2011 © Cut X-Ray mod
 Author     : SkyLoader
-------------------------------------------------]]

local flag = false

local flag_dead,flad_circle = false,false
local time,time_c = 0,time_global() + 500

function is_killed()
	if not db.actor:alive() then
		if not flag_dead and not flad_circle then
		local dir=device().cam_dir
		db.actor:set_actor_direction(dir.x,dir.y,0) --'выравниваем направление взгляда, чтобы было падение более реалистичным
 		level.add_cam_effector("camera_effects\\surge_02.anm", 2532, false, "")
		level.add_pp_effector("surge_fade.ppe", 2011, false)
		level.disable_input()
		time = time_global()
		flag_dead=true
		end
		if flag_dead then
			if time_global()-time>3000 then
			level.enable_input()
			level.add_pp_effector("deadcity_wake.ppe", 4467, false)
			flag_dead=false
			flad_circle=true
			end
		end

		if flad_circle then
			if time_c<=time_global() then
			level.remove_pp_effector(4467)
			level.add_pp_effector("deadcity_wake.ppe", 4467, false)
			time_c = time_global() + 500
			end
		end
	end
end
```

Но оказалось что level.add_pp_effector и вроде level.add_cam_effector не работают после смерти.

4. Не плохобы было добавить в скрипты метод возвращающий интенсивность дождя под которым находится гг. Пологаю что нечто подобное уже есть на примере шейдера Raindrops (тк он проявляется только тогда когда на камеру попадает дождь), но оно не доступно скриптам. В дальнейшем уже скриптами из этого можно было-бы сделать радиоактивные осадки к примеру или добавить звук ударяющихся капель об капюшон или стекло севы.

5. Как на счет добавить вот такой шейдер https://www.youtube.com/watch?v=ZkY-6WuPrgM&t=6s 
он-же visor reflections он-же visor.h (из шейдеров anomaly) В сочетании с каплями на стекле и отражениями на асфальте смотрелось-бы очень не плохо. Впринципе можно было-бы выделить под него отдельную консольную консольную команду, этого было-бы достаточно. Ну а потом тем кому нужно уже смогли-бы написать простой скрипт который включал-бы этот эффект когда одет определенный костюм и выключал-бы его когда костюм снят.

**Comment by xrSimpAI** [comment]
1. Он и так есть. `sun_shafts_intensity` называется.
2. Это всё субъективно.
3. Нет.
4. Он даже в оригинальном движке есть - `level.rain_factor()`.
5. Я уже не раз отвечал, что если бы это был на 100% шейдерный эффект, я бы его может и добавил, но к нему нужно под 100 мб текстур. Это меня не устраивает.

**Comment by bigmack2304** [comment]
1. Про это не знал, спасибо.

4. Ну level.rain_factor() возвращает значение дождя который вообще на всей локации, а я имел ввиду то как узнать это конкретно для актора, ведь если я буду находится в помещении то это значение будет 0 независимо от того какой силы там будет идти дождь.

5. Вот [тут](https://www.amk-team.ru/forum/topic/12673-sheydery/?do=findComment&comment=1074306) выкладывали шейдеры с этим эффектом, там все в месте 6мб весит. Правда это под 2й рендер, но может будет интересно.

**Comment by xrSimpAI** [comment]
Ну вместе с rain_factor можно попробовать `actor:is_actor_outdoors()` но это не особо надежная штука. По уму надо бы это всё переделать но мне лень.

А насчёт шейдеров - ну там прям старье какое-то. Я имел ввиду те шейдеры что в аномали. А эти походу ты можешь и без движка адаптировать, судя по тому что там походу околованильный двиг юзается, не думаю что в него именно для шлемов что-то могли добавить. Консольную команду для включения какую-то разве что.

UPD: нет, там походу всё через команду `r2_aa`, которая была бесполезна в оригинале, у нас вообще вырезана.

**Comment by sadasdasdqwedasdasdwqe** [comment]
почему бы  не попробовать , сделать возможность менять разные аддоны по типу прицелов и глушителей,  На одной кости,
не создавая разные худы модели  одного оружия

**Reply by I-AmDead** [reply]
К чему это?

**Comment by Balavnik** [comment]
Предлагаю добавить возможность включения OGSE-слотов по отдельности.

**Reply by PRXPHET** [reply]
Типа отдельно под нож, отдельно под бинокль и т.п.?

**Reply by Balavnik** [reply]
Именно. Т.е. в features чтобы была возможность по отдельности включать, например slot_knife = true врубает нож.

**Reply by joye-ramone** [reply]
включи все слоты, а тебе что тебе не надо не используй в конфига.

**Reply by Balavnik** [reply]
Муторно очень. Для них ведь всех dragdrop в inventory_new создавать надо, а  если только dragdrop_knife сделать, то будет вылет по причине не найденных dragdrop_torch, pda и т.д.

**Reply by imcrazyhoudini** [reply]
Да не муторно это, возьми готовый пример с модов да и всё, во всяком случае можно выше взять тот же слот пистолета, скопировать и переименовать как надо и скрыть, пару минут дела.

**Reply by joye-ramone** [reply]
> Муторно очень. Для них ведь всех dragdrop в inventory_new создавать надо, а если только dragdrop_knife сделать, то будет вылет по причине не найденных dragdrop_torch, pda и т.д.

а перепиливать двиг потому что тебе лень скопировать 4 раза xml и поменять имя не муторно?)
